### PR TITLE
Fix for Windows errors caused by files left unclosed

### PR DESCRIPTION
**Pull request by k1o0**

### CODE REVIEW

File: mtscomp.py [hunk 1]
```diff
@@ -804,23 +804,26 @@ def __del__(self):
...
+            # Check the dtype and shape match.
+            assert chunk.dtype == expected.dtype
+            assert chunk.shape == expected.shape
+            if np.issubdtype(chunk.dtype, np.integer):
+                # For integer dtypes, check that the array are exactly equal.
+                assert np.array_equal(chunk, expected)
+            else:
+                # For floating point dtypes, check that the array are almost equal
+                # (diff followed by cumsum does not yield exactly the same floating point numbers).
+                assert np.allclose(chunk, expected, atol=CHECK_ATOL)
```
Comment (rossant) [inline mtscomp.py:824]: Will this try block hide any exception raised in it? If so perhaps we should add a `logger.error()` in the `except` block?

Comment (k1o0) [reply mtscomp.py:824]: Because there's no except block it will continue to raise any error after the finally block

**Comment by k1o0** [comment]
> Thanks for the PR. Does it fix all of the Windows issues?

All the mtscomp tests run on my Windows machine after this fix.  There were nessesary changed to be made to the spikeglx module in ibllib too.  After releasing this we need to [merge this branch](https://github.com/int-brain-lab/ibllib/compare/develop...spikeglx).

File: mtscomp.py (hunk 1 above)
Comment (rossant) [reply mtscomp.py:824]: ok!

**Review by rossant — APPROVED** [review]

**Pull request merged**